pkg/frontend/frontend.go

```diff
@@ -223,243 +228,146 @@ func NewFrontend(ctx context.Context,
+				r.Get("/serialconsole", f.getAdminOpenShiftClusterSerialConsole)
 
-	s = r.
-		Path("/admin/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}/drainnode").
-		Subrouter()
+				r.Get("/clusterdeployment", f.getAdminHiveClusterDeployment)
 
-	s.Methods(http.MethodPost).HandlerFunc(f.postAdminOpenShiftClusterDrainNode).Name("postAdminOpenShiftClusterDrainNode")
+				r.Post("/redeployvm", f.postAdminOpenShiftClusterRedeployVM)
 
-	s = r.
-		Path("/admin/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}/clusterdeployment").
-		Subrouter()
+				r.Post("/stopvm", f.postAdminOpenShiftClusterStopVM)
 
-	s.Methods(http.MethodGet).HandlerFunc(f.getAdminHiveClusterDeployment).Name("getAdminHiveClusterDeployment")
+				r.Post("/startvm", f.postAdminOpenShiftClusterStartVM)
 
-	s = r.
-		Path("/admin/versions").
-		Subrouter()
+				r.Post("/upgrade", f.postAdminOpenShiftUpgrade)
 
-	s.Methods(http.MethodGet).HandlerFunc(f.getAdminOpenShiftVersions).Name("getAdminOpenShiftVersions")
-	s.Methods(http.MethodPut).HandlerFunc(f.putAdminOpenShiftVersion).Name("putAdminOpenShiftVersions")
+				r.Get("/skus", f.getAdminOpenShiftClusterVMResizeOptions)
 
-	// Operations
-	s = r.
-		Path("/providers/{resourceProviderNamespace}/operations").
-		Queries("api-version", "").
-		Subrouter()
+				r.Post("/resize", f.postAdminOpenShiftClusterVMResize)
 
-	s.Methods(http.MethodGet).HandlerFunc(f.getOperations).Name("getOperations")
+				r.Post("/reconcilefailednic", f.postAdminReconcileFailedNIC)
 
-	s = r.
-		Path("/subscriptions/{subscriptionId}").
-		Queries("api-version", "2.0").
-		Subrouter()
+				r.Post("/cordonnode", f.postAdminOpenShiftClusterCordonNode)
 
-	s.Methods(http.MethodPut).HandlerFunc(f.putSubscription).Name("putSubscription")
-}
+				r.Post("/drainnode", f.postAdminOpenShiftClusterDrainNode)
+			})
+		})
 
-func (f *frontend) setupRouter() *mux.Router {
-	r := mux.NewRouter()
-	r.Use(f.logMiddleware.Log)
-	r.Use(f.m.Metrics)
-	r.Use(middleware.Panic)
-	r.Use(middleware.Headers)
-	r.Use(f.validateMiddleware.Validate)
-	r.Use(middleware.Body)
-	r.Use(middleware.SystemData)
-
-	r.NotFoundHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Header().Set("Content-Type", "application/json")
-		api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeNotFound, "", "The requested path could not be found.")
+		// Operations
+		r.Route("/providers/{resourceProviderNamespace}", func(r chi.Router) {
+			r.Get("/{resourceType}", f.getAdminOpenShiftClusters)
+		})
 	})
 
-	r.NotFoundHandler = f.authMiddleware.Authenticate(r.NotFoundHandler)
-
-	unauthenticated := r.NewRoute().Subrouter()
-	f.unauthenticatedRoutes(unauthenticated)
+	r.Put("/subscriptions/{subscriptionId}", f.putSubscription)
```

Comment:
do we have other routes that are relying on it? I have not found conflicting routes on this and no reference to the api-version in the function, it seems to be some leftover from time I cannot remember because I was not there :D

Reply:
Fair enough, me neither. So lets see if it breaks something. Lets be ready to fix it quick if anything happens.